### PR TITLE
Create roles

### PR DESCRIPTION
**Pull request by Thommond**
Created the createRoles.php file.

Added the PHP code to allow admin to create role names and access levels via the form which then check the database to see if that role exists, if not then adds it to the database.

Added list of current roles already in the database and displayed it on the page.

Made the common-function.php file to hold functions that will be used through out the project.

Updated all view files to have check_session() function.

### CODE REVIEW

**Comment by Thommond** [comment]
This needs to be updated for session checking, so it should not be added until that update is completed.

File: common-functions.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  $db_link = mysqli_connect("localhost", "root", '', "retire");
+
+  if ($db_link == false) {
+
+    die("ERROR: Can not connect to database" . mysqli_connect_error());
+  }
+
+  $sql = "SELECT id, email
+          FROM users
+          WHERE id '$the_id'";
```
Comment (BrendanHorst) [inline common-functions.php]: shouldn't there be an `=`here?

File: common-functions.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      $result = mysqli_fetch_row(mysqli_query($db_link, $sql));
+
+      // Check if session data matches database
+      if ($result[0] != $_SESSION['id'] or $result[1] != $_SESSION['email'])  {
+        header('Location: http://localhost/Retire-Inspired/views/errors/forbidden.php');
+      }
+
+      echo 'bob is cool';
+
+      mysqli_close($db_link);
```
Comment (BrendanHorst) [inline common-functions.php]: This should not be inside the `if`, as if the query fails the connection will not close

File: common-functions.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  if (mysqli_query($db_link, $sql)) {
+
+      $result = mysqli_fetch_row(mysqli_query($db_link, $sql));
+
+      // Check if session data matches database
+      if ($result[0] != $_SESSION['id'] or $result[1] != $_SESSION['email'])  {
+        header('Location: http://localhost/Retire-Inspired/views/errors/forbidden.php');
+      }
+
+      echo 'bob is cool';
```
Comment (BrendanHorst) [inline common-functions.php]: who is bob

File: common-functions.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  if ($db_link == false) {
+
+    die("ERROR: Can not connect to database" . mysqli_connect_error());
+  }
+
+  $sql = "SELECT id, email
+          FROM users
+          WHERE id '$the_id'";
+
+  if (mysqli_query($db_link, $sql)) {
```
Comment (BrendanHorst) [inline common-functions.php:49]: I feel like this `if` needs an `else` with an error page.  If the query fails the page will load like normal without checking the user id or email

**Review by BrendanHorst — CHANGES_REQUESTED** [review]
Just a few notes on the check_session function

File: common-functions.php (hunk 4 above)
Comment (Thommond) [reply common-functions.php:49]: I have added the else statement as well.

**Review by BrendanHorst — APPROVED** [review]

**Pull request merged**